src/api.v2/controllers/gem2sController.js

```diff
@@ -9,8 +9,8 @@ const ExperimentParent = require('../model/ExperimentParent');
 
 const logger = getLogger('[Gem2sController] - ');
 
-const runGem2s = async (req, res) => {
-  const { experimentId } = req.params;
+const runGem2s = async (stateMachineParams, authorization) => {
```

Comment:
The functions exported by the controllers should be functions that handle params `req` `res`. They are supposed to handle routes directly.

If this function is no longer doing that, then it should be moved over to a different place (probably where all the pipeline stuff is, perhaps we can even merge this with other functions? idk).

I noticed this same situation with seurat and subset controller.

Reply:
Personally I'm leaning more towards separating the params and passing them individually if there are not many of them (3 or less perhaps?), but whatever you prefer

Reply:
The reason I'm leaning towards separating the params is that having them in an object kind of hides them, and it's not worth it when it's not many params

Reply:
I understand it is more tractable to separate the parameters, but the reason I went with passing an entire object is because when the `/retry` endpoint is called the object containing the last parameters that is in the db can be directly passed no matter what the function is. Afterwards each function (`runSubset`, `runSeurat`, `runGem2s`) extracts their necessary parameters.

Otherwise, in the retry endpoint I would have to use a switch statement and for each function I would extract the proper parameters. I really do not like that solution as it is not scalable and I think would lower the code quality. 

imo we should leave it like this

Reply:
I understand it is more tractable to separate the parameters, but the reason I went with passing an entire object is because when the `/retry` endpoint is called the object containing the last parameters that is in the db can be directly passed no matter what the function is. Afterwards each function (`runSubset`, `runSeurat`, `runGem2s`) extracts their necessary parameters.

Otherwise, in the retry endpoint I would have to use a switch statement and for each function I would extract the proper parameters. I really do not like that solution as it is not scalable and I think would lower the code quality. 

imo we should leave it like this

Reply:
Ok, although I wouldn't suggest avoiding a `switch` when possible (there are usually better ways to define things). 

For example, it could be stored in your `const stateMachineToPipeline` object which already stores functions as `[SUBSET_PROCESS_NAME]: (...) => { blablabla; executeSubsetPipeline(...) }`.

I see your point though, what do you think of having `runSubset`, `runSeurat`, `runGem2s` assert that all the necessary properties in these parameter objects are defined? And throw an error if they aren't. That should help bring out to light any possible bugs we might introduce in the future/possible incorrect stored parameters.

Reply:
If you like the idea of the `assert`, since you are going to use it in 3 different files I would suggest checking if there isn't an already accessible `assert` function. 

If there isn't, you could add a small function, something like this:
```
assert = (condition) => {
    if (!condition) throw new Error("Assertion failed")
}
```

Reply:
Why not use the function `validateRequest` with a RetryRequest.yaml like we do for most requests in the API? Seems like the cleanest solution to ensure the correct params are passed after being fetched from the database.

Reply:
hmm I hadn't thought of that because it isn't a request, just a collection of parameters